### PR TITLE
Add container goal samples

### PR DESCRIPTION
**Pull request by ddgenome**
Add simple, parallel goals, multi-platform, caching, callback, and
volume container goal samples.

Update @atomist/sdm-core to version with container goal support.

### CODE REVIEW

File: lib/sdm/container/3cache.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                container("node", {
+                    containers: [{
+                        args: ["sh", "-c", "npm install && npm test"],
+                        env: [{ name: "NODE_ENV", value: "development" }],
+                        image: "node:12.4.0",
+                        name: "npm",
+                    }],
+                    output: [{
+                        classifier: "node-modules",
+                        pattern: { directory: "node_modules" },
```
Comment (twicksell) [inline lib/sdm/container/3cache.ts:62]: Would it be worth having an example on how to work with the output captured here? Maybe a simple example of log parsing or capturing mocha / maven test results?

Comment (ddgenome) [reply lib/sdm/container/3cache.ts:62]: This output captures files on the file system to a persistent cache for the goal set, not log/stdout/stderr output. This output is used as input to the next goal. Is that sufficient or would you like an example that processes stdout/stderr from the container? Currently the stdout/stderr of the container is just sent to the goal's progress log.

**Review by cdupuis — APPROVED** [review]
This looks really good.

File: lib/sdm/container/3cache.ts (hunk 1 above)
Comment (twicksell) [reply lib/sdm/container/3cache.ts:62]: I spotted using the classifier to push data to the next goal, but I was thinking along the lines of accessing that output from within the SDM. Right now it looks like I can only get to that output data from another container task within the same goalset. I'm looking for something like a callback to be executed against the output of the container goal, which could modify/enrich the goal's result. 

A simple example might something like: Parse the output of the test results upon goal completion, and add the number of tests passed/failed to the message of the goal result.

Comment (cdupuis) [reply lib/sdm/container/3cache.ts:62]: Right now the log output is accessible from the `logInterpreter`. What are the kind of things you'd like to table to do when parsing the log output?

Comment (cdupuis) [reply lib/sdm/container/3cache.ts:62]: > A simple example might something like: Parse the output of the test results upon goal completion, and add the number of tests passed/failed to the message of the goal result.

I think that might be possible today by using the `ProjectListener` stuff we got. I'll see if I can create an example for that.

Comment (ddgenome) [reply lib/sdm/container/3cache.ts:62]: If you want to modify the value stored in the cache, you can create an intermediate goal that performs the modification and stores the updated value in the cache for the next goal.

If you want to parse/enrich the stdout/stderr, you can use the `logInterpreter` like @cdupuis said to get the content of the log and report on it via other means, e.g., a chat message, or provide an alternative `ProgressLogFactory` to the SDM that actually modifies the log in line.

**Pull request merged**